Gems/Multiplayer/Code/Source/Components/NetworkRigidBodyComponent.cpp

```diff
@@ -28,20 +28,28 @@ namespace Multiplayer
 
     void NetworkRigidBodyComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
     {
+        NetworkRigidBodyComponentBase::GetProvidedServices(provided);
         provided.push_back(AZ_CRC_CE("NetworkRigidBodyService"));
     }
 
     void NetworkRigidBodyComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
     {
+        NetworkRigidBodyComponentBase::GetRequiredServices(required);
         required.push_back(AZ_CRC_CE("PhysXRigidBodyService"));
     }
 
     void NetworkRigidBodyComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent)
     {
+        NetworkRigidBodyComponentBase::GetDependentServices(dependent);
         dependent.push_back(AZ_CRC_CE("TransformService"));
         dependent.push_back(AZ_CRC_CE("PhysXRigidBodyService"));
     }
 
+    void NetworkRigidBodyComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        NetworkRigidBodyComponentBase::GetIncompatibleServices(incompatible);
+    }
+
```

Comment:
Provided and Incompatible are actually somewhat redundant with the base here. The base provides NetworkRigidBodyComponent while this provides NetworkRigidBodyService. We need to standardize around one and then remove the superfluous push from NetworkRigidBodyComponent.

Reply:
Digging through code a bit more, what you can do here is simply not define `GetProvidedServices` and `GetIncompatibleServices` and then correct the following in NetworkCharacterComponent

```
void NetworkCharacterComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
{
    incompatible.push_back(AZ_CRC_CE("NetworkRigidBodyService"));
}
```
to use the proper name for NetworkRigidBody and to call the base.

Reply:
All the autocomponent provide a service with their own name. That could be useful, but typically services that conflict with one another will have a more generic name since they are both providing the same thing. For example, NetworkRigidBody and NetworkCharacter both provide "PhysXRigidBodyService"

While it's redundant, there's no problem in providing both. Will anyone actually ever be incompatible with a "NetworkRigidBodyComponent" specifically? Probably not. They will care more about "PhysXRigidBodyService"

What might be better than having jinja blindly provide a service with the same name as the component, might be to make a field in the autocomponent.xml to explicitly provide/depend/require a service. Thoughts?

Reply:
Two things here, first a component shouldn't key itself twice in dependency management. We should remove `AZ_CRC_CE("NetworkRigidBodyService")`. 

Second, as far as explicitly calling the base. I would try profiling both approaches and see what's faster here. In one case, the base is explicitly called. In the other, the base has to be resolved which has to be dealt with under the hood. I would suspect the latter but I don't have data to support that.

On the subject of jinja, AutoComponent jinja templating has the ability to specify dependency relationships so far through Component Constraints but it implies a convention that's not particularly enforceable outside of usage of other AutoComponents. That's a limitation I believe we just have to accept.

In the case of NetworkCharacterComponent, you could probably do something like this in the xml.

```
<ComponentRelation Constraint="Incompatible" HasController="true" Name="NetworkRigidBodyComponent" Namespace="Multiplayer" Include="Multiplayer/Components/NetworkRigidBodyComponent.h" />
```

Reply:
Ah, I see what's up... if derived never creates its own services methods, the base is used.
Okay, I've update the PR, i think it's cleaner now. The incompatibility is defined straight in the XML.